lib/swagger/index.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+/*
+ * Dependencies
+ */
+var fs = require('fs');
+var yaml = require('js-yaml');
+var path = require('path');
+var swagger = require('swagger-tools');
+
+var log = require('../logger');
+var requestErrorHandler = require('./requestErrorHandler');
+var responseErrorHandler = require('./responseErrorHandler');
+
+/**
+ * Swagger binds the swagger doc to route handlers, sets up validation, and UI
+ *
+ * @param app
+ * @param options
+ */
+module.exports = function initSwagger (app, options) {
+
+  try {
+    // Implementor swagger defined routes
+    var swaggerDoc = yaml.safeLoad(fs.readFileSync(path.resolve(options.callerPath, options.swaggerConfig.filePath)));
+
+    swagger.initializeMiddleware(swaggerDoc, function (middleware) {
+
+      // Interpret Swagger resources and attach metadata to request - must be first in swagger-tools middleware chain
+      app.use(middleware.swaggerMetadata());
+
+      // Validate Swagger requests
+      app.use(middleware.swaggerValidator({
+        validateResponse: true
+      }));
+      
+      //  Custom error handler to convert swagger request errors into API friendly json variants
+      app.use(requestErrorHandler);
+
+      // Route validated requests to appropriate controller
+      app.use(middleware.swaggerRouter({
+        controllers: options.swaggerConfig.controllers
+      }));
+
+      //  Custom error handler to convert swagger standard application/html response errors into API friendly json variants
+      app.use(responseErrorHandler);
```

Comment:
Does this also work with Error handling middleware for custom errors thrown from the controllers?

Reply:
Can you please elaborate? Where would you envisage the need for extra custom error handling beyond the swagger response validation if one chose to validate responses through the swagger specification.

Reply:
I think I might have just worked out what you are asking...

Do you mean would it override any app.use(errorMiddleWare) that was added after app.use(micro) was called from an implementing application?

eg:

app.use(micro({options}));
app.use(customErrorHandler);

app.listen(...)

Reply:
Yes, I am not too familiar with swagger-tools, it could potentially wire our application differently to how we do error handling with express.

Reply:
It currently would cause issues the way I've done it as I added a catch all response in that handler as that's how I roll. I'll remove it and update the tests to confirm it still works with custom handlers as per your original comment.

Reply:
I've updated this.

One has to listen to the app.once('swagger:routes:registered', () ={}) event to do so. Obviously this only needs to be done if binding to swagger; otherwise everything works exactly as before if not.